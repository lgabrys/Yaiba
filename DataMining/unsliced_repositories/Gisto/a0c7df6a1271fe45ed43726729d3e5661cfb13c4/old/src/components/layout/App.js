import React from 'react';
import PropTypes from 'prop-types';
import styled, { createGlobalStyle } from 'styled-components';
import * as Mousetrap from 'mousetrap';
import { withRouter } from 'react-router-dom';

import { isElectron } from 'utils/electron';

import SuperSearch from 'components/layout/SuperSearch';
import KeyBindings from 'components/layout/KeyBindings';

// if (process.env.NODE_ENV === 'development') {
//   require('mimic');
// }

const GlobalStyle = createGlobalStyle`
  ::-webkit-scrollbar {
    width: 5px;
    height: 5px;
  }
  
  ::-webkit-scrollbar-track {
    background: #fff;
  }
  
  ::-webkit-scrollbar-thumb {
    background: ${(props) => props.theme.baseAppColor};
  }
  
  a:hover,
  span:hover {
    transition: all .3s ease-in-out;
  }
  
  *:focus {
    outline-color: ${(props) => props.theme.baseAppColor};
  }
`;

const AppWrapper = styled.div`
  display: flex;
  background: #fff;
  flex-direction: column;
  justify-content: flex-start;
  height: 100vh;
  color: ${(props) => props.theme.baseAppColor};
`;

export class App  extends React.Component {
  state = {
    showSuperSearch: false,
    showKeyBindings: false
  };

  componentDidMount() {
    Mousetrap.bind(['command+f', 'ctrl+f'], this.toggleSuperSesrch);
    Mousetrap.bind(['?'], this.toggleKeyBindings);
    Mousetrap.bind(['esc'], this.toggleAllOff);

    // Disallow to drop files
    document.addEventListener('dragover', (event) => event.preventDefault());
    document.addEventListener('dragenter', (event) => event.preventDefault());
    document.addEventListener('dragleave', (event) => event.preventDefault());
    document.addEventListener('drop', (event) => event.preventDefault());

    if (isElectron) {
      const { ipcRenderer } = require('electron');

      // Route if 'routeTo' event sent from main
      ipcRenderer.on('routeTo', (event, routeTo) => {
        this.props.history.push(routeTo);
      });
    }
  }

  componentWillUnmount() {
    Mousetrap.unbind(['command+f', 'ctrl+f'], this.toggleSuperSesrch);
    Mousetrap.unbind(['?'], this.toggleKeyBindings);
    Mousetrap.unbind(['esc'], this.toggleAllOff);
  }

  toggleSuperSesrch = () => {
    this.setState((prevState) => ({
      showSuperSearch: !prevState.showSuperSearch
    }));
  };

  toggleKeyBindings = () => {
    this.setState((prevState) => ({
      showKeyBindings: !prevState.showKeyBindings
    }));
  };

  toggleAllOff = () => {
    this.setState({
      showKeyBindings: false,
      showSuperSearch: false
    });
  };

  render() {
    return (
      <AppWrapper>
        <GlobalStyle/>
        { this.props.children }
        { this.state.showSuperSearch && (
          <SuperSearch toggleSuperSesrch={ () => this.toggleSuperSesrch() }/>
        ) }
        { this.state.showKeyBindings && (<KeyBindings/>) }
      </AppWrapper>
    );
  }
}

App.propTypes = {
  children: PropTypes.node,
  history: PropTypes.object
};

export default withRouter(App);
