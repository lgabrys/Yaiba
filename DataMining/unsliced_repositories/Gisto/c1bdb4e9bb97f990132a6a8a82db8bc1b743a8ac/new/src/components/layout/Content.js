import React from 'react';
import { HashRouter as Router, Route, Switch } from 'react-router-dom';
import styled from 'styled-components';

import DashBoard from 'components/layout/content/DashBoard';
import Settings from 'components/layout/content/Settings';
import Snippet from 'components/layout/content/Snippet';
import About from 'components/layout/content/About';
import NewSnippet from 'components/layout/content/NewSnippet';

const ContentWrapper = styled.div`
  flex: 1;
  padding: 20px;
  background: ${(props) => props.theme.bg};
  overflow: auto;
  position: relative;
`;

export const Content = () => (
  <ContentWrapper>
    <Router>
      <Switch>
        <Route exact path="/" component={ DashBoard }/>
        <Route path="/about" component={ About }/>
        <Route path="/settings" component={ Settings }/>
        <Route path="/snippet/:id" component={ Snippet }/>
        <Route path="/new" component={ NewSnippet }/>
      </Switch>
    </Router>
  </ContentWrapper>
);

Content.propTypes = {};

export default Content;
