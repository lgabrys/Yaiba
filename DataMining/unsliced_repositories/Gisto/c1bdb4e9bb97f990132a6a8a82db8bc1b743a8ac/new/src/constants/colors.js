import { getSetting } from 'utils/settings';
import { tint } from 'polished';

export const baseAppColor = getSetting('color', '#3F84A8');
export const borderColor = '#ccc';
export const disabledColor = borderColor;
export const lightBorderColor = '#f2f2f2';
export const lightText = '#fff';
export const textColor = '#555';
export const boxShadow = textColor;
export const colorDanger = 'tomato';
export const colorSuccess = 'cadetblue';
export const colorWarning = '#e0b01e';
export const headerColor = lightText;
export const headerBgLightest = tint(0.25, baseAppColor);
export const bg = tint(0.10, baseAppColor);

export const theme = {
  baseAppColor: getSetting('color', '#3F84A8'),
  borderColor: '#ccc',
  disabledColor: '#ccc',
  lightBorderColor: '#f2f2f2',
  lightText: '#fff',
  textColor: '#555',
  boxShadow: '#555',
  colorDanger: 'tomato',
  colorSuccess: 'cadetblue',
  colorWarning: '#e0b01e',
  headerColor: '#fff',
  headerBgLightest: tint(0.25, getSetting('color', '#3F84A8')),
  bg: tint(0.10, getSetting('color', '#3F84A8'))
};
