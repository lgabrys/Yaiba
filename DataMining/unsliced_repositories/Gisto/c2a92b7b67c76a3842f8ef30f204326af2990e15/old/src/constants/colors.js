import { getSetting } from 'utils/settings';
import { tint } from 'polished';

export const baseAppColor = getSetting('color') || '#3F84A8';
export const borderColor = '#ccc';
export const disabledColor = borderColor;
export const lightBorderColor = '#f2f2f2';
export const lightText = '#fff';
export const textColor = '#555';
export const boxShadow = textColor;
export const colorDanger = 'tomato';
export const colorSuccess = 'cadetblue';
export const colorWarning = '#e0b01e';
export const headerColor = lightText;
export const headerBgLightest = tint(0.25, baseAppColor);
export const bg = tint(0.10, baseAppColor);
