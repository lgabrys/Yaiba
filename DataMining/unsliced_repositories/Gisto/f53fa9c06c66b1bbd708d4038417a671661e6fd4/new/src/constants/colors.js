import Color from 'color';

export const borderColor = '#ccc';
export const lightBorderColor = '#f2f2f2';
export const lightText = '#fff';
export const textColor = '#555';
export const boxShadow = textColor;
export const baseAppColor = '#3F84A8';
export const colorRegular = baseAppColor;
export const colorDanger = 'tomato';
export const colorSuccess = 'cadetblue';
export const colorWarning = '#e0b01e';
export const colorInfo = '#1c93d2';
export const borderColorLight = borderColor;
export const headerColor = lightText;
export const headerBgLightest = Color(baseAppColor).lighten(1.1).string();
export const bg = Color(baseAppColor).lighten(0.95).string();
