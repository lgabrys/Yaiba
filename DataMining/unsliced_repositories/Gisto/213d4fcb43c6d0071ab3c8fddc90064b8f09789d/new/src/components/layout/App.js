import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { baseAppColor } from 'constants/colors';

const AppWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  height: 100vh;
  color: ${baseAppColor};
`;

const App = (props) => (
  <AppWrapper>{ props.children }</AppWrapper>
);

App.propTypes = {
  children: PropTypes.node
};

export default App;
