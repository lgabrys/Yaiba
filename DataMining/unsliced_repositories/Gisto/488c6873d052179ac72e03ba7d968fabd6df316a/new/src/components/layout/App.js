import React from 'react';
import PropTypes from 'prop-types';
import styled, { injectGlobal } from 'styled-components';
import * as Mousetrap from 'mousetrap';
import { withRouter } from 'react-router-dom';

import { baseAppColor } from 'constants/colors';
import { isElectron } from 'utils/electron';

import SuperSearch from 'components/layout/SuperSearch';
import KeyBindings from 'components/layout/KeyBindings';

// eslint-disable-next-line no-unused-expressions
injectGlobal`
  ::-webkit-scrollbar {
    width: 5px;
    margin: 0 20px 0 0;
  }
  
  ::-webkit-scrollbar-track {
    background: #fff;
  }
  
  ::-webkit-scrollbar-thumb {
    background: ${baseAppColor};
  }
`;

const AppWrapper = styled.div`
  display: flex;
  background: #fff;
  flex-direction: column;
  justify-content: flex-start;
  height: 100vh;
  color: ${baseAppColor};
`;

export class App  extends React.Component {
  state = {
    showSuperSearch: false,
    showKeyBindings: false
  };

  componentDidMount() {
    Mousetrap.bind(['shift shift', 'ctrl+f'], this.toggleSuperSesrch);
    Mousetrap.bind(['?'], this.toggleKeyBindings);
    Mousetrap.bind(['esc'], this.toggleAllOff);

    // Disallow to drop files
    document.addEventListener('dragover', (event) => event.preventDefault());
    document.addEventListener('dragenter', (event) => event.preventDefault());
    document.addEventListener('dragleave', (event) => event.preventDefault());
    document.addEventListener('drop', (event) => event.preventDefault());

    if (isElectron) {
      const { ipcRenderer } = require('electron');

      // Route if 'routeTo' event sent from main
      ipcRenderer.on('routeTo', (event, routeTo) => {
        this.props.history.push(routeTo);
      });
    }
  }

  componentWillUnmount() {
    Mousetrap.unbind(['shift shift', 'ctrl+f'], this.toggleSuperSesrch);
    Mousetrap.unbind(['?'], this.toggleKeyBindings);
    Mousetrap.unbind(['esc'], this.toggleAllOff);
  }

  toggleSuperSesrch = () => {
    this.setState((prevState) => ({
      showSuperSearch: !prevState.showSuperSearch
    }));
  };

  toggleKeyBindings = () => {
    this.setState((prevState) => ({
      showKeyBindings: !prevState.showKeyBindings
    }));
  };

  toggleAllOff = () => {
    this.setState({
      showKeyBindings: false,
      showSuperSearch: false
    });
  };

  render() {
    return (
      <AppWrapper>
        { this.props.children }
        { this.state.showSuperSearch && (
          <SuperSearch toggleSuperSesrch={ () => this.toggleSuperSesrch() }/>
        ) }
        { this.state.showKeyBindings && (<KeyBindings/>) }
      </AppWrapper>
    );
  }
}

App.propTypes = {
  children: PropTypes.node,
  history: PropTypes.object
};

export default withRouter(App);
