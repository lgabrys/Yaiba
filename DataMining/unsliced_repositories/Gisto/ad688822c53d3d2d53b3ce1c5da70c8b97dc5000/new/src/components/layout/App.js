import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { baseAppColor } from 'constants/colors';
import * as Mousetrap from 'mousetrap';
import SuperSearch from 'components/layout/SuperSearch';

const AppWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  height: 100vh;
  color: ${baseAppColor};
`;

export class App  extends React.Component {
  state = {
    showSuperSearch: false
  };

  componentDidMount() {
    Mousetrap.bind(['shift shift', 'ctrl+f'], this.toggleSuperSesrch);
  }
  componentWillUnmount() {
    Mousetrap.unbind(['shift shift', 'ctrl+f'], this.toggleSuperSesrch);
  }

  toggleSuperSesrch = () => {
    this.setState({
      showSuperSearch: this.state.showSuperSearch = !this.state.showSuperSearch
    });
  };

  render() {
    return (
      <AppWrapper>
        { this.props.children }
        { this.state.showSuperSearch && (
          <SuperSearch toggleSuperSesrch={ () => this.toggleSuperSesrch() }/>
        ) }
      </AppWrapper>
    );
  }
}

App.propTypes = {
  children: PropTypes.node
};

export default App;
