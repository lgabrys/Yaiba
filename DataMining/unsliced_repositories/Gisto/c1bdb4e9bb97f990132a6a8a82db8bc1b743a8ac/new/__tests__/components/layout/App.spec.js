import { App } from 'components/layout/App';
import * as theme from 'constants/colors';

const propSetup = (props = {}) => ({
  theme,
  ...props
});

const setup = (props) => shallow(<App { ...propSetup(props) }/>);

describe('COMPONENTS - <App>', () => {
  test('render App', () => {
    const component = setup();

    expect(component).toMatchSnapshot();
  });
});
