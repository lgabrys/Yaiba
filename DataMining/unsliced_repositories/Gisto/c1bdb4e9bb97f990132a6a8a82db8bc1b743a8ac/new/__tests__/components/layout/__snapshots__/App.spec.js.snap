// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENTS - <App> render App 1`] = `
<ForwardRef>
  <GlobalStyleComponent
    suppressMultiMountWarning={false}
  />
</ForwardRef>
`;
